src/Settings/Builders/RunTest.hs

```diff
@@ -23,23 +28,27 @@ runTestBuilderArgs = builder RunTest ? do
             | pkg <- pkgs, isLibrary pkg, pkg /= rts, pkg /= libffi ]
 
     flav <- expr flavour
-    rtsways <- rtsWays flav
+    rtsways <- expr $ testRTSSettings
     libways <- libraryWays flav
     let hasRtsWay w = elem w rtsways
         hasLibWay w = elem w libways
         debugged = ghcDebugged flav
-
-    withNativeCodeGen <- expr ghcWithNativeCodeGen
-    withInterpreter   <- expr ghcWithInterpreter
-    unregisterised    <- getFlag GhcUnregisterised
-    withSMP           <- expr ghcWithSMP
+    hasProfiling  <- expr $ fmap stringToBool (testSetting TestGhcProfiled) 
+    hasDynamic    <- expr $ fmap stringToBool (testSetting TestGhcDynamic)
+    hasDebugged   <- expr $ fmap stringToBool (testSetting TestGhcDebugged)
+    hasDynamicByDefault <- expr $ fmap stringToBool (testSetting TestGhcDynamicByDefault)
+    withNativeCodeGen <- expr $ fmap stringToBool (testSetting TestGhcWithNativeCodeGen)
+    withInterpreter   <- expr $ fmap stringToBool (testSetting TestGhcWithInterpreter)
+    unregisterised    <- expr $ fmap stringToBool (testSetting TestGhcUnregisterised)
+    withSMP           <- expr $ fmap stringToBool (testSetting TestGhcWithSMP)
```

Comment:
I _think_ it's safe to say that we might want a shorter name for `expr $ fmap stringToBool (testSetting ...)` (or just the `fmap stringToBool (testSetting ...)` part?) =)